charts/consul/templates/mesh-gateway-deployment.yaml

```diff
@@ -240,7 +240,11 @@ spec:
             -service-partition={{ .Values.global.adminPartitions.name }} \
             {{- end }}
             -log-level={{ default .Values.global.logLevel }} \
-            -log-json={{ .Values.global.logJSON }}
+            -log-json={{ .Values.global.logJSON }} \
+            {{- if (and .Values.global.metrics.enabled .Values.global.metrics.enableGatewayMetrics) }}
+            -telemetry-prom-scrape-path={{ .Values.connectInject.metrics.defaultPrometheusScrapePath }} \
+            -telemetry-prom-merge-port={{ .Values.connectInject.metrics.defaultMergedMetricsPort }}
```

Comment:
Why do we need this for gateways? There should be no merged metrics and the port should come from the service's `envoy_prometheus_bind_addr`

Reply:
`-telemetry-prom-scrape-path` is needed because consul-dataplane does not have a default for it. You are correct about the `-telemetry-prom-merge-port` and I have removed it.